[8.4] docs: add 8.4 known issue (#139437)
elastic/kibana: docs/CHANGELOG.asciidoc

Review the following information about the {kib} 8.4.0 release.

[float]
[[known-issue-8.4.0]]
=== Known issues

> If you have alerting rules with configured snooze notifications, do not upgrade {kib} to 8.4.0.

Reviewer: ```suggestion
If you have alerting rules configured to snooze notifications, do not upgrade {kib} to 8.4.0.
```

Author: Can you check what I changed it to and let me know if you like it?

Reviewer: Yes, that sounds good!

> If you have alerting rules with configured snooze notifications, do not upgrade {kib} to 8.4.0.

Author: ```suggestion
If you have alerting rules that have been snoozed, do not upgrade {kib} to 8.4.0.
```

> If you have alerting rules with configured snooze notifications, do not upgrade {kib} from 8.3.0 to 8.4.0.

Author: ```suggestion
If you have alerting rules with configured snooze notifications, do not upgrade {kib} to 8.4.0.
```

> To verify if your alerting rules have snooze notifications, open the main menu, then click **Stack Management** -> **Rules and Connectors**.

Author: ```suggestion
To determine if you have snoozed alerting rules, open the main menu, then click **Stack Management** -> **Rules and Connectors**.
```

Filter the rule list by selecting **View** -> **Snoozed**.
For each space, cancel the snooze for all affected rules before you upgrade.

> To identify snoozed rules in Spaces using **Dev Tools**, run the following query to search for the `is_snoozed_until` string:

Author: ```suggestion
To identify snoozed rules in all Spaces using **Dev Tools**, run the following query and then search for the `is_snoozed_until` string:
```

> To identify snoozed rules in all Spaces using **Dev Tools**, run the following query and then search for the `is_snoozed_until` string:
> 
> [source,console]
> ----
> GET kbn:api/alerting/rules/_find
> ----

Author: ```suggestion
To identify snoozed rules in all Spaces using **Dev Tools**, run the following query:

[source,console]
----
GET /.kibana/_search
{
  "query": {
    "exists": {
      "field": "alert.isSnoozedUntil"
    }
  }
}
----
```

> When you upgrade {kib} to 8.4.0, and your alerting rules have snooze notifications, you will receive the following error message:

Reviewer: ```suggestion
When you upgrade {kib} to 8.4.0, and you have alerting rules configured to snooze notifications, you will receive the following error message:
```

> When you upgrade {kib} from 8.3.0 to 8.4.0, and your alerting rules have snooze notifications, you receive the following error message:

Author: ```suggestion
When you upgrade {kib} to 8.4.0, and your alerting rules have snooze notifications, you will receive the following error message:
```

Review the following information about the {kib} 8.4.0 release.

[float]
[[known-issue-8.4.0]]
=== Known issues

> Do not upgrade Kibana from 8.3 to 8.4 if you have any alerting rules with a snooze notification configured.

Reviewer: ```suggestion
Do not upgrade Kibana from 8.3 to 8.4 if you have any alerting rules that are snoozed.
```

`snooze notification` seems confusing to me because snoozing will silence any notifications. Not sure if the above suggestion is the best tho

Author: 👍 Thanks! I'll update.

You can identify snoozed rules in {kib} by navigating to **Stack Management** -> **Rules and Connectors**.
Filter the rule list by selecting **View** -> **Snoozed**.

> Cancel the snooze for all affected rules before upgrading. This must be done for each Space.

Reviewer 2: ```suggestion
For each space, cancel the snooze for all affected rules before you upgrade.
```

> You can also identify snoozed rules in any Space in {kib} **Dev Tools**.

Reviewer 2: ```suggestion
To identify snoozed rules in Spaces using **Dev Tools**, run the following query to search for the `is_snoozed_until` string:
```

> Run the following query and searching for the string `is_snoozed_until`.

Reviewer 2: ```suggestion
```

[source,console]
----
GET kbn:api/alerting/rules/_find

> ----

Reviewer: Alternatively, we could run an ES query via the console to only get snoozed rules:

```
GET .kibana/_search
{
  "query": {
    "exists": {
      "field": "alert.isSnoozedUntil"
    }
  }
}
```

Reviewer: Another reason to use the ES query is the kbn command will only list rules in the default space. Users would need to run GET kbn:s/<spaceId>/api/alerting/rules/_find to find rules in each of their spaces

Author: Does this return results from all spaces?

Author: Perfect!

Reviewer: Yes, the ES query will work across spaces

Author: Updated!

> If you attempt to upgrade Kibana from 8.3 to 8.4 with snoozed rules you will receive the following error message.

Reviewer 2: ```suggestion
When you upgrade {kib} from 8.3.0 to 8.4.0, and your alerting rules have snooze notifications, you receive the following error message:
```

[source,text]
----
FATAL  Error: Unable to complete saved object migrations for the [.kibana] index.
----

> You must restore your previous version and cancel any snoozed rules before retrying the upgrade.

Reviewer 2: ```suggestion
Before you retry the upgrade, restore your previous version, then cancel any alerting rules with snooze notifications.
```
